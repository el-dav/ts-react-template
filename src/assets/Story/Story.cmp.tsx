import React from 'react';

import Theme from '~/assets/Theme';
import Store from '~/assets/Store';

interface Props {
  isConnected?: boolean;
}

const Story: React.SFC<Props> = ({ children, isConnected }) =>
  isConnected ? (
    <Store>
      <Theme>{children}</Theme>
    </Store>
  ) : (
    <Theme>{children}</Theme>
  );

export default Story;
