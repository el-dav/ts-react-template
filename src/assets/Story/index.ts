import Story from './Story.cmp';

export default Story;
