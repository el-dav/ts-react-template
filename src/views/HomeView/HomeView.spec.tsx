import React from 'react';
import renderer from 'react-test-renderer';

import Story from 'assets/Story';

import HomeViewCmp from './HomeView.cmp';
import HomeViewCnt from './HomeView.cnt';

describe('HomeView', () => {
  it('renders the Component correctly', () => {
    const tree = renderer
      .create(
        <Story isConnected>
          <HomeViewCmp />
        </Story>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Container correctly', () => {
    const tree = renderer
      .create(
        <Story isConnected>
          <HomeViewCnt />
        </Story>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
