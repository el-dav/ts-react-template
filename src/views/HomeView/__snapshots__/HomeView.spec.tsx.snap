// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renders the Component correctly 1`] = `
<div
  className=""
>
  Home
</div>
`;

exports[`HomeView renders the Container correctly 1`] = `
<div
  className=""
>
  Home
</div>
`;
